### ISSUE TITLE
Add \hidetype to hide type aliases' definitions

### ISSUE THREAD

**Issue by cjdb** [doxygen/doxygen#10129]
Some type aliases are defined in terms of implementation details that I'd prefer not to expose to users. Instead, I intend to describe the alias' in prose, which will be much more understandable.

For example:

```cpp
/// Alias for `std::incrementable_traits<T>::value_type` unless `std::iterator_traits<T>` has an explicit
/// specialisation for `T`, and `value_type` is a member alias of that specialisation.
/// \hidetype
template<class T>
using iter_value_t = typename __detail::choose_iter_value<remove_cvref_t<T>>::type;
```
would hide `typename __detail::choose_iter_value<remove_cvref_t<T>>::type`, similar to how `\hideinitializer` hides variable initialisers.

**Comment by albert-github** [comment]
I think there might be more places where also "initializations / aliases" take place.

@doxygen what do you think
1. extra command `\hidetype` (and `\showtype`) as suggested
2. use `\hideinitializer` / `\showinitializer` also for `using`
3. use `\hideinitializer` / `\showinitializer` with an option (e.g. `\hideinitializer{using}`
4. regarding `MAX_INITIALIIZER_LINES`
   1. use it also for `using`
   2. separate option for `using`

**Comment by doxygen** [comment]
@albert-github I would opt for options 2. and 4.i.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10140

**Comment by albert-github** [comment]
This proposed pull request of theis issue (#10140) is obsolete due to the commit [73874de](https://github.com/doxygen/doxygen/commit/73874def7998419647a6281ec4b39c4261d4d18f) for the issue https://github.com/doxygen/doxygen/issues/10334

As the fix is not yet part of on official release, please don't close the issue as this will be done at the moment of an official release

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.10.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
